Sources/ArgumentParser/Usage/HelpGenerator.swift

```diff
@@ -144,8 +144,15 @@ internal struct HelpGenerator {
     var optionElements: [Section.Element] = []
     /// Used to keep track of elements already seen from parent commands.
     var alreadySeenElements = Set<Section.Element>()
+    
+    var commandsToShowHelp = [ParsableCommand.Type]()
+    if let commandType = commandStack.last, !commandType.includeSuperCommandInHelp {
+        commandsToShowHelp.append(commandType)
+    } else {
+        commandsToShowHelp = commandStack
+    }
```

Comment:
You should be able to just grab the top of the command stack this way...
```suggestion
    guard let commandType = commandStack.last else {
        return []
    }
```

...and then remove the following `for`-`in` loop and de-dent its contents.